shared/health/src/main/java/io/pravega/shared/health/impl/HealthServiceUpdaterImpl.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health.impl;
+
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.google.common.util.concurrent.Service;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.concurrent.Services;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.HealthServiceUpdater;
+import io.pravega.shared.health.HealthService;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Slf4j
+@RequiredArgsConstructor
+public class HealthServiceUpdaterImpl extends AbstractScheduledService implements HealthServiceUpdater {
+
+    /**
+     * The {@link Health} the daemon should report in the case it has not started querying the {@link HealthService}.
+     */
+    private static final Health INITIAL_HEALTH = Health.builder().build();
+
+    /**
+     * Represents the most recent {@link Health} information provided by the {@link ScheduledExecutorService}.
+     */
+    private final AtomicReference<Health> latest = new AtomicReference<Health>(INITIAL_HEALTH);
+
+    /**
+     * The {@link HealthService} associated with this {@link HealthServiceUpdater}.
+     */
+    private final HealthService service;
+
+    /**
+     * The underlying {@link ScheduledExecutorService} used to executor the recurring service-level {@link Health} check.
+     */
+    private final ScheduledExecutorService executorService = ExecutorServiceHelpers.newScheduledThreadPool(1, "health-service-updater", Thread.MIN_PRIORITY);
```

Comment:
Where do you shut this down?

Reply:
I thought that by overriding `executor()` it would close it down for use, but I see now that its actually the opposite..